### ISSUE TITLE
CUDA_ERROR_INVALID_CONTEXT on cupy.cuda.compiler.compile_with_cache

### ISSUE THREAD

**Issue by vadimkantorov** [cupy/cupy#940]
I'm using cupy to compile kernels to further use with PyTorch. The code below fails with CUDA_ERROR_INVALID_CONTEXT, because CUDA seems not initialized for cupy.

If I put a dummy `cupy.array([])` call before `compile_with_cache`, everything works smoothly.

```python
import string
import collections
import cupy
import torch

CUDA_NUM_THREADS = 1024
GET_BLOCKS = lambda N:  (N + CUDA_NUM_THREADS - 1) // CUDA_NUM_THREADS
Stream = collections.namedtuple('Stream', ['ptr'])

def f(name = 'inc'):
    kernel = b'''
        extern "C"
        __global__ void ${name}(${Dtype}* output, const ${Dtype}* input, int n)
        {
            int i = blockIdx.x * blockDim.x + threadIdx.x;
            if (i >= n)
                return;
            const ${Dtype} &x = input[i];
            output[i] = input[i] + 1;
        }
        '''
    module = cupy.cuda.compiler.compile_with_cache(string.Template(kernel).substitute(Dtype = 'float', name = name))
    return module.get_function(name)

# cupy.array([1, 2, 3]) # uncommenting this line fixes the issue
func = f()
data = torch.cuda.FloatTensor([1,2,3])

print(data)
func(block=(CUDA_NUM_THREADS,1,1),
      grid=(GET_BLOCKS(data.numel()),1,1),
      args=[data.data_ptr(), data.data_ptr(), data.numel()],
      stream=Stream(ptr=torch.cuda.current_stream().cuda_stream))
print(data)
```

```
Traceback (most recent call last):
  File "bar.py", line 26, in <module>
    func = f()
  File "bar.py", line 22, in f
    module = cupy.cuda.compiler.compile_with_cache(string.Template(kernel).substitute(Dtype = 'float', name = name))
  File ".../python2.7/site-packages/cupy/cuda/compiler.py", line 149, in compile_with_cache
    mod.load(cubin)
  File "cupy/cuda/function.pyx", line 175, in cupy.cuda.function.Module.load
  File "cupy/cuda/function.pyx", line 176, in cupy.cuda.function.Module.load
  File "cupy/cuda/driver.pyx", line 141, in cupy.cuda.driver.moduleLoadData
  File "cupy/cuda/driver.pyx", line 72, in cupy.cuda.driver.check_status
cupy.cuda.driver.CUDADriverError: CUDA_ERROR_INVALID_CONTEXT: invalid device context
```

**Comment by kmaehashi** [comment]
Thanks for reporting.
This issue should be fixed after https://github.com/cupy/cupy/pull/916 got merged (currently under review).

**Comment by kmaehashi** [comment]
Fixed via #916.
